docs/src/dev/io/graphbinary.asciidoc

```diff
@@ -0,0 +1,771 @@
+- `{value}` is the two's complement of the `BigInteger`.
+
+Example values of the two's complement `{value}`:
+
+- `00`: Integer 0.
+- `01`: Integer 1.
+- `127`: Integer 7f.
+- `00 80`: Integer 128.
+- `ff`: Integer -1.
+- `80`: Integer -128.
+- `ff 7f`: Integer -129.
+
+==== Byte
+
+An unsigned 8-bit integer.
+
+==== ByteBuffer
+
+Format: `{length}{value}`
+
+Where:
+
+- `{length}` is an `Int` representing the amount of bytes contained in the value.
+- `{value}` sequence of bytes.
+
+==== Short
+
+Format: 2-byte two's complement integer.
+
+==== Boolean
+
+Format: A single byte containing the value `0x01` when it's `true` and `0` otherwise.
+
+==== TextP
+
+Format: `{predicate}{values_length}{value_0}...{value_n}`
+
+Where:
+
+- `{name}` is a String.
+- `{values_length}` is an `Int` describing the amount values.
+- `{value_i}` is a fully qualified typed value composed of `{type_code}{type_info}{value_flag}{value}`.
+
+==== TraversalStrategy
+
+Format: `{strategy_class}{configuration}`
+
+Where:
+
+- `{strategy_class} is a `Class` that is of type `TraversalStrategy`.
+- `{configuration} is a `Map` of data used to configure the strategy that will be given to a `TraversalStrategy` `create(Configuration)` method.
+
+==== BulkSet
+
+Format: `{length}{item_0}...{item_n}`
+
+Where:
+
+- `{length}` is an `Int` describing the length of the `BulkSet`.
+- `{item_0}...{item_n}` are the items of the `BulkSet`. `{item_i}` is sequence of a fully qualified typed value composed of
+`{type_code}{type_info}{value_flag}{value}` followed by the "bulk" which is a `Long` value.
+
+If the implementing language does not have a `BulkSet` object to deserialize into, this format can be coerced to a
+`List` and still be considered compliant with Gremlin. Simply create "expand the bulk" by adding the item to the `List`
+the number of times specified by the bulk.
+
+==== Tree
+
+Format: `{length}{item_0}...{item_n}`
+
+Where:
+
+- `{length}` is an `Int` describing the length of the items.
+- `{item_0}...{item_n}` are the items of the `Tree`. `{item_i}` is composed of `{key}` is a fully-qualified type value
+followed by a `{Tree}`.
+
+==== Metrics
+
+Format: `{id}{name}{duration}{counts}{annotations}{nested_metrics}`
+
+Where:
+
+- `{id}` is a `String` representing the identifier.
+- `{name}` is a `String` representing the name.
+- `{duration}` is a `Long` describing the duration in nanoseconds.
+- `{counts}` is a `Map` composed by `String` keys and `Long` values.
+- `{annotations}` is a `Map` composed by `String` keys and a value of any type.
+- `{nested_metrics}` is a `List` composed by `Metrics` items.
+
+==== Custom
+
+A custom type, represented as a blob value.
+
+Type Info: `{name}{custom_type_info}`
+
+Where:
+
+- `{name}` is a `String` containing the implementation specific text identifier of the custom type.
+- `{custom_type_info}` is a `ByteBuffer` representing the additional type information, specially useful
+for complex custom types.
+
+Value format: `{blob}`
```

Comment:
Sure, but that's not what I meant. My problem was simply that I wasn't sure how the value format and the type info are combined to form the full serialized value, but I just read the first paragraph of this document again where this is explained with:
```
{type_code}{type_info}{value_flag}{value}
```
My confusion probably just came because I read this document top to bottom and the custom type is the first where a special type info is used. That shouldn't be a problem however for someone else who jumps directly from the start of the document to this type (or for someone with a better memory^^)